docs/architecture/different_tool_versions.md

```diff
@@ -0,0 +1,130 @@
+Being fully flexible in the versions of all supported tools would lead to a combinatorics explosion.
+So, the configuration mechanism needs to be quite flexible as well.
+
+When triggering an ORT run, users must be able to specify information about version constraints in a generic way.
+With labels, ORT (and ORT Server) already supports a mechanism to provide arbitrary data that is available during the whole run.
+So it would be possible to define a specific prefix, e.g., _transport:_, for labels that are intended to be evaluated by transport implementations.
+
+When invoking a worker, the Orchestrator could extract all labels starting with this prefix and add them to the message that triggers the worker.
+For this, the `Message` class would have to be extended.
+
+NOTE:
+This approach would have the additional benefit that other dynamic configuration information specific for the transport implementation in use could be handled.
+One example of such additional properties could be CPU and memory limits for Kubernetes pods that could be adapted to the expected result size.
+
+The transport implementation has its own configuration which is currently static.
+This could be extended to support special placeholders in the values of configuration properties that are substituted with corresponding label values.
+That way, the behavior of a transport can be influenced in a generic way.
+
+Let's make this more concrete with an example using the Kubernetes transport implementation.
+The following diagram shows the data flow from a request to trigger an ORT run until the start of the container image for the Analyzer as a Kubernetes job.
+
+```mermaid
+sequenceDiagram
+    actor user as User
+    participant api as API
+    participant script as Parameter Validation Script
+    participant orchestrator as Orchestrator
+    participant transport as Kubernetes Transport
+    participant kubernetes as Kubernetes
+        
+    user ->> api: Trigger run<br/>javaVersion=17
+    api ->> script: Validate parameters
+    script ->> orchestrator: Add label<br/>transport:ort-server-analyzer-image-tag=java17-latest
+    orchestrator ->> transport: Add transport property<br/>kubernetes:imageTagSuffix=java17-latest
+    transport ->> kubernetes: Start job with image<br/>ort-server-analyzer:java17-latest
+```
+
+The flow begins with the user sending a request to trigger a run for a specific repository.
+As described under [Parameters when triggering an ORT run](../../website/docs/guides/parameter-validation-script.md), the API to trigger requests can be customized using a validation script.
+In this example, the script supports a _javaVersion_ parameter in the generic parameters of the Analyzer job configuration.
+The value is the version of the JDK that needs to be installed in the container image running the Analyzer step.
+
+NOTE:
```

Comment:
See above.

Reply:
I have removed the whole paragraph, it does not match the new diagram.